egui_demo_lib/src/apps/demo/widget_gallery.rs

```diff
@@ -171,7 +171,15 @@ impl WidgetGallery {
         ui.end_row();
 
         ui.add(doc_link_label("DragValue", "DragValue"));
-        ui.add(egui::DragValue::new(scalar).speed(1.0));
+        ui.add(egui::DragValue::new(scalar).speed(1.0).clamp_range(0..=360));
+        ui.end_row();
+
+        ui.add(doc_link_label("StepButtons", "StepButtons"));
+        ui.add(
+            egui::StepButtons::new(scalar)
+                .step(1.0)
+                .clamp_range(0..=360),
```

Comment:
This doesn't actually show the value - isn't it more common for one to want to show the value between the -/+ buttons?

At this point it feels like this is just:

``` rust
if ui.button("+").clicked() { value += 1 }
if ui.button("-").clicked() { value -= 1 }
value = value.clamp(0, 360);
```

with very little extra value added.

See https://www.nngroup.com/articles/input-steppers/ for some examples of input steppers.

I think for this widget to motivate it's own existence, it should also display the value (between the -/+ buttons), and if the user wants finer control than that, they can just write it out themselves (like I did above).

What do you think?

Reply:
Yes, I'm afraid you're right. I still think that step buttons would be useful but maybe this is not the way to do it. Maybe it would be possible to keep the widget atomic and still have the step buttons on it. But I'm closing this issue for now.